docs/user_guide/event_system.md

```diff
@@ -0,0 +1,147 @@
+| Queue | Queue type changed ||
+| User/group | Limit configured for a user ||
+| User/group | Limit configured for a group ||
+| User/group | Limit removed for a user ||
+| User/group | Limit removed for a group ||
+| User/group | Resource usage increased for a user ||
+| User/group | Resource usage increased for a group | Only sent if group limit is active |
+| User/group | Group tracking got associated with a user| Only sent if group limit is active |
+| User/group | Group tracking removed from a user| Only sent if group limit is active |
+
+`*` a predicate is a plugin located in the default scheduler. They're responsible for functionality like node selectors, affinity, anti-affinity, etc. Yunikorn runs the predicates for every allocation to see if it fits the candidate node.
+
+# In-memory storage of historical events
+
+Yunikorn stores a previously generated event objects in a ring buffer. Once it gets full, oldest elements are overwritten. The default size of the buffer is 100000.
+On a busy cluster this might not be enough. The size can be changed in the configmap by using the key `event.ringBufferCapacity`.
+Keep in mind that increasing the size results in higher memory usage and longer GC pause times. Events that are put in the ring buffer are not removed, only overwritten.
+If the size of the buffer gets smaller after a config change, the oldest entries that do not fit are discarded.
+
+Every event is assigned an ID starting from 0. This can be used to fetch them based on ID.
+
+Yunikorn currently does not offer a solution which stores the generated events in a persistent storage. However, this might change in the upcoming releases.
+
+# Retrieving events
+
+The REST API provides two ways to see the generated events.
+
+## Batch REST endpoint
+
+The batch endpoint is available at `/ws/v1/events/batch`. If not defined, the start ID is 0 and a maximum of 10000 events are returned. This can be changed by defining the "start" and "count" URL query parameters.
+
+See [batch interface](../api/scheduler.md#batch-events) for details.
+ 
+## Streaming REST endpoint
+
+The streaming endpoint is available at `/ws/v1/events/stream`. Unlike the batch interface, which closes the connection after the query, streaming keeps the HTTP connection open. There is no idle timeout, so as long as the connection is stable, it is never closed. New events are sent immediately to active clients. The URL parameter "count" is also accepted; it tells Yunikorn how many recent events we want to fetch that were generated before the connection.
+
+Since this approach uses more resources inside Yunikorn, the number of streaming connections is limited, but the limits are configurable. See [streaming](../api/scheduler.md#event-stream) for details. Existing connections are not closed after configuration change.
+
+The active streaming connections are also shown in the state dump.
+
+# Event structure
+
+A Yunikorn event has the following fields:
```

Comment:
Does it include the change of YUNIKORN-2454?

Reply:
That's separate from `si.EventRecord`.

Reply:
My point was: should we remind users the record introduced by YUNIKORN-2454? The record is different from `si.EventRecord` so user should "parse" it by another way.

Reply:
I think it's only relevant in the REST API docs. However, we still need changes on that front (eg. at least the starting ID need to be included). So we can fix that later.